### PR TITLE
feat(design-system): ErrorState (WIP)

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
ErrorState is not existing yet but is required for a feature.

**What is the chosen solution to this problem?**
Implement it using draft guidelines https://www.figma.com/file/o5kR1lrKR9he56PyyfWaTa/Error-States?node-id=0%3A1

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+- The entire app
+- The entire current screen
+- A specific element on the screen
+
+Depending on the scope, the error state may display directly on the screen, in dialogs, or in toasters.
+
+## Zoning
+
+<FigmaImage src="" alt="Zoning image for ErrorState" />
```
Comment (Benrajalu) [inline packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx:33]: @AnLefevre Do you think you could build a quick zoning for this? :)

File: packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## Style
+
+<FigmaImage src="" alt="Styles for ErrorStates" />
```
Comment (Benrajalu) [inline packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx:37]:
```suggestion
<FigmaImage src="" alt="Styles for ErrorStates" />
```
@sgangnant You have no URL for this? 🤔

Comment (sgangnant) [reply packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx:37]: Not sure what this sould be, would that be the "Specs" node ? https://www.figma.com/file/o5kR1lrKR9he56PyyfWaTa/Error-States?node-id=104%3A5159

File: packages/design-system/src/components/WIP/ErrorState/ErrorState.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import ErrorIllustration from './illutstrations/ErrorIllustration';
+
+import styles from './ErrorState.module.scss';
+
+type ErrorStatePropTypes = {
+	title: string;
+	description: string;
+	action?: Omit<ButtonPrimaryPropsType<'M'>, 'size'>;
+	link?: ReactElement | string;
```
Comment (Benrajalu) [inline packages/design-system/src/components/WIP/ErrorState/ErrorState.tsx:15]:
```suggestion
	link?: ReactElement | LinkProps;
```
We must try to ensure that there is a link in there. `string` would be useless.

By checking if the prop is a function or an object, you can either display the `<Link />` component or the provided node.

**Review by romainseb — APPROVED** [review]

File: packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx (hunk 2 above)
Comment (Benrajalu) [reply packages/design-system/src/components/WIP/ErrorState/ErrorState.stories.mdx:37]: https://www.figma.com/file/o5kR1lrKR9he56PyyfWaTa/Error-States?node-id=104%3A5159 this node should do!

**Review by Benrajalu — APPROVED** [review]

**Review by tlnd-mhuchet — APPROVED** [review]
❤️

**Pull request merged**